packages/client/src/runtime/core/types/Extensions.ts

```diff
@@ -149,7 +149,7 @@ type DynamicModelExtensionFluentApi<TypeMap extends TypeMapDef, M extends Proper
   [K in keyof TypeMap['model'][M]['payload']['objects']]:
     <A>(args?: Exact<A, Path<TypeMap['model'][M]['operations'][P]['args']['select'], [K]>>) =>
       & PrismaPromise<Path<DynamicModelExtensionFnResultBase<TypeMap, M, { select: { [P in K]: A } }, P>, [K]> | Null>
-      & DynamicModelExtensionFluentApi<TypeMap, (TypeMap['model'][M]['payload']['objects'][K] & {})['name'], P, Null>
+      & DynamicModelExtensionFluentApi<TypeMap, (TypeMap['model'][M]['payload']['objects'][K] & {})['name'], P, Null | Select<TypeMap['model'][M]['payload']['objects'][K], null>>
```

Comment:
Extensions suffered from a part of the problem, `null`s were not forwarded when chaining on optional relations.